compiler/src/dotty/tools/dotc/quoted/reflect/ReflectionCompilerInterface.scala

```diff
@@ -1224,6 +1224,11 @@ class ReflectionCompilerInterface(val rootContext: Context) extends CompilerInte
   def Type_select(self: Type)(sym: Symbol)(using Context): Type =
     self.select(sym)
 
+  def Type_appliedTo(self: Type)(targs: List[Type]): Type =
+    self match
+      case self: Types.HKTypeLambda => self.instantiate(targs)
+      case _ => Types.AppliedType(self, targs)
```

Comment:
I meant to say this should be:
```suggestion
    self.appliedTo(targs)
```
Since TypeApplications#appliedTo handles a bunch of cases: https://github.com/lampepfl/dotty/blob/0d22c74ed6418482193a7190a30f24bbf88b4b3d/compiler/src/dotty/tools/dotc/core/TypeApplications.scala#L285